src/acc/libsmm_acc/libcusmm/libcusmm.cpp

```diff
@@ -207,41 +211,15 @@ int libcusmm_process_d(int *param_stack, int stack_size, CUstream stream, int m,
 
         static bool cpu_fallback = false;
 
-        // Add lock to table of locks and initialize it
-        // (Some serialization here)
+        // OpenMP version: make sure we JIT the kernel only once for each (m, n, k) (Some serialization here)
 #if defined _OPENMP
```

Comment:
Without OpenMP aren't you now recompiling every time?

Reply:
No, there is a [lookup](https://github.com/shoshijak/dbcsr/blob/fix/207/src/acc/libsmm_acc/libcusmm/libcusmm.cpp#L210) for the kernel a few lines above.